### DISCUSSION TITLE
App database backups should not include files in subfolders

### DISCUSSION THREAD

**Discussion by Gys** [pocketbase/pocketbase#5747] [Q&A] [answered]
Hi!

I keep some files in subfolders of the folder holding the database. Those files are rather large. I noticed the backups are including those files. Looking through the pb code I see this:
https://github.com/pocketbase/pocketbase/blob/0b304351830a6b8d1254f9ed69f9991c0247243b/tools/archive/create.go#L17
And:
https://github.com/pocketbase/pocketbase/blob/master/core/base_backup.go#L61

So pb explicitly excludes some internal subfolders and includes all others. I wonder why, because the `https://pocketbase.io/docs/api-backups/#create-backup` page says 'Creates a new app data backup'. So I assumed it did that: backup *only* the database.

I think the 'database backup' should do be limited to database files, because those are important and dynamic by nature. No need or requirement to include other unknown app specific files who happen to be in one or more subfolders. Maybe even limit the backup very specifically to `data.*` and `logs.*` files and nothing else.

If there is a need to include other, non-db files in the backup then maybe make that into a setting. 

I came across this 'feature' when I tried to unzip a backup. It failed because in the new location was no equally named subfolder. If there had been such subfolder, the files in there would have been overwritten silently!

**Comment by ganigeorgiev** [comment, marked as answer]
You can argue whether it should or not but it is how it works at the moment.

The reasoning behind it is that you can have custom app specific files in `pb_data` that they could be required for your application to work and not including them could result in restore failure.

In any case with the upcoming v0.23.0 there is now [`OnBackupCreate` hook](https://pkg.go.dev/github.com/pocketbase/pocketbase@v0.23.0-rc8/core#BaseApp.OnBackupCreate) that allow specifying an `Exclude` entries list.

The backups functionality will be also further refactored to handle the case of mixed `pb_data` content mount points (https://github.com/pocketbase/pocketbase/issues/4647) but that will be after the v0.23.0 release.

**Reply by Gys** [reply]
I still think an INclude list would be make more sense ;) but making the exclude list accessible is a good solution!
